code/modules/mob/living/carbon/xenomorph/castes/widow/abilities_widow.dm

```diff
@@ -0,0 +1,291 @@
+	/// List of beams to be removed on obj_destruction
+	var/list/obj/effect/ebeam/beams
+	/// Beam var
+	var/obj/effect/ebeam/beam
+	layer = ABOVE_ALL_MOB_LAYER
+	anchored = TRUE
+
+/// Humans caught get beamed and registered for proc/check_dist, aoe_leash also gains increased integrity for each caught human
+/obj/structure/xeno/aoe_leash/Initialize(mapload)
+	. = ..()
+	var/list/obj/effect/ebeam/beams = list()
+	for(var/mob/living/carbon/human/victim in view(leash_radius, loc))
+		beam = (beam(victim, "beam_heavy", 'icons/obj/items/projectiles.dmi', INFINITY, INFINITY))
+		beams += beam
+		RegisterSignal(victim, COMSIG_MOVABLE_MOVED, .proc/check_dist)
+		obj_integrity = obj_integrity + integrity_increase
+		if(obj_integrity > max_integrity)
+			obj_integrity = max_integrity
+	if(!beams)
+		return INITIALIZE_HINT_QDEL
+
+/// To remove beams after the leash_ball is destroyed
+/obj/structure/xeno/aoe_leash/obj_destruction()
+	. = ..()
+	if(!beams)
+		return
+	for(beam in beams)
+		qdel(beam)
+
+/// Humans caught in the aoe_leash will be pulled back if they leave it's radius
+/obj/structure/xeno/aoe_leash/proc/check_dist(datum/leash_victim, atom/oldloc)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/human/victim = leash_victim
+	if(get_dist(victim, oldloc) >= leash_radius)
+		victim.Move(oldloc)
+
+/// This is so that xenos can remove leash balls
+/obj/structure/xeno/aoe_leash/attack_alien(mob/living/carbon/xenomorph/X, damage_amount = X.xeno_caste.melee_damage, damage_type = BRUTE, damage_flag = "", effects = TRUE, armor_penetration = 0, isrightclick = FALSE)
+	if(X.status_flags & INCORPOREAL)
+		return
+	X.visible_message(span_xenonotice("\The [X] starts tearing down \the [src]!"), \
+	span_xenonotice("We start to tear down \the [src]."))
+	if(!do_after(X, 2 SECONDS, TRUE, X, BUSY_ICON_GENERIC))
+		return
+	if(!istype(src))
+		return
+	X.do_attack_animation(src, ATTACK_EFFECT_CLAW)
+	X.visible_message(span_xenonotice("\The [X] tears down \the [src]!"), \
+	span_xenonotice("We tear down \the [src]."))
+	playsound(src, "alien_resin_break", 25)
+	take_damage(max_integrity)
+
+// ***************************************
+// *********** Spiderling Section
+// ***************************************
+
+/datum/action/xeno_action/create_spiderling
+	name = "Birth Spiderling"
+	ability_name = "birth_spiderling"
+	mechanics_text = " Spawn a spiderling directly under you"
+	action_icon_state = "spawn_hugger" // temporary until I get my own icons
+	plasma_cost = 1 // increase later
+	cooldown_timer = 1 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_LEASH_BALL
+	/// List of all our spiderlings
+	var/list/mob/living/carbon/xenomorph/spiderling/spiderlings
+	/// Max amount of spiderligns
+	var/max_spiderlings = 5
+
+/datum/action/xeno_action/create_spiderling/New(Target)
+	. = ..()
+	spiderlings = list()
+
+/// The action to create spiderlings
+/datum/action/xeno_action/create_spiderling/action_activate()
+	. = ..()
+	if(spiderlings.len >= max_spiderlings)
```

Comment:
personal note we usually use length() instead of .len because it accounts for null and is about the same if not slightly faster